de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/process/C_BankStatement_ImportAttachment.java

```diff
@@ -0,0 +1,124 @@
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.process.RunOutOfTrx;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.util.api.Params;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_AD_AttachmentEntry;
+import org.compiere.model.I_I_BankStatement;
+
+public class C_BankStatement_ImportAttachment extends JavaProcess implements IProcessPrecondition
+{
+	/*
+	Having DataImportConfigId hardcoded is fine.
+	We could use a sysconfig, but what shall we store there? the ID? the InternalName? (internal name is editable by user). That won't change/fix anything.
+	If you have a better suggestion, please ping me.
+	 */
+	public static final DataImportConfigId HARDCODED_BANK_STATEMENT_DATA_IMPORT_REPO_ID = DataImportConfigId.ofRepoId(540009);
+
+	@Param(parameterName = I_AD_AttachmentEntry.COLUMNNAME_AD_AttachmentEntry_ID, mandatory = true)
+	private AttachmentEntryId p_AD_AttachmentEntry_ID;
+
+	private static final String BANK_STATEMENT_MUST_BE_IN_PROGRESS_MSG = "Bank Statement must be in progress.";
+
+	private final IMsgBL iMsgBL = Services.get(IMsgBL.class);
+	private final AttachmentEntryService attachmentEntryService = SpringContextHolder.instance.getBean(AttachmentEntryService.class);
+	private final transient DataImportService dataImportService = SpringContextHolder.instance.getBean(DataImportService.class);
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable(@NonNull final IProcessPreconditionsContext context)
+	{
+		if (context.isNoSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		if (!context.isSingleSelection())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNotSingleSelection();
+		}
+
+		final I_C_BankStatement selectedBankStatement = context.getSelectedModel(I_C_BankStatement.class);
+		final DocStatus docStatus = DocStatus.ofCode(selectedBankStatement.getDocStatus());
+		if (!docStatus.isDraftedOrInProgress())
+		{
+			return ProcessPreconditionsResolution.reject(iMsgBL.getTranslatableMsgText(BANK_STATEMENT_MUST_BE_IN_PROGRESS_MSG));
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@RunOutOfTrx
+	@Override
+	protected String doIt() throws Exception
+	{
+		final AttachmentEntryDataResource data = attachmentEntryService.retrieveDataResource(getAttachmentEntryId());
+
+		dataImportService.importData(DataImportRequest.builder()
+				.data(data)
+				.dataImportConfigId(HARDCODED_BANK_STATEMENT_DATA_IMPORT_REPO_ID)
+				.clientId(getClientId())
+				.orgId(getOrgId())
+				.userId(getUserId())
+				.additionalParameters(computeImportProcessParams())
+				.build());
+
+		return MSG_OK;
+	}
+
+	private Params computeImportProcessParams()
+	{
+		final I_C_BankStatement bankStatement = Services.get(IBankStatementDAO.class).getById(getRecord_ID());
```

Comment:
introduce `Services.get(IBankStatementDAO.class)` as a field.
Note u will need it in checkPreconditionsApplicable too

Reply:
done